### DISCUSSION TITLE
Spawn method stopped working randomly?

### DISCUSSION THREAD

**Discussion by Tofaa2** [Minestom/Minestom#1576] [Help] [answered]
So i was making a little fun minigame server and i needed mobs to spawn. I have a janky method to spawn them which isnt ideal but it works and performance isnt that important rn.

```java
    public void spawn(final Pos position) {
        try {
            final PassiveMob mob = mobClass.getConstructor().newInstance();
            mob.setInstance(SAOCraft.getWorld());
            mob.spawn();
            mob.teleport(position);
        }
        catch (Exception e) {
            e.printStackTrace();
        }
    }
    ```
    That is the method It gets called in the class below:
    
    ```
        private static class PassiveCommand extends SAOCraftCommand {
        public PassiveCommand() {
            super(Rank.ADMINISTRATOR, "passive");
            final ArgumentEnum<PassiveMobs> mobArgument = ArgumentType.Enum("mob", PassiveMobs.class);

            this.addSyntax((sender, context) -> {
                final PassiveMobs mob = context.get(mobArgument);
                if (!(sender instanceof SAOCraftPlayer player)) {
                    sender.sendMessage(ChatColor.translate("<red>You must be a player to use this command!"));
                    return;
                }
                mob.spawn(player.getPosition());
                player.sendMessage(ChatColor.translate("<green>Spawned " + mob.name().toLowerCase() + "!"));
            }, mobArgument);
        }
    }
    ```
    
    This used to work just fine but all of a sudden just stopped spawning the mob. The green Spawned message gets sent but i cannot see the mob anymore. Any ideas?

**Comment by GreatWyrm** [comment]
The spawn method in the Entity seems to be just for overrides (could probably be replaced with EntitySpawnEvent with little issue).

Can you try `mob.setInstance(SAOCraft.getWorld(), position);` instead?

**Reply by Tofaa2** [reply]
Yea i know the Entity#spawn() method does nothing i just put it there fore safe measure, this piece of code used to work i've modified nothing so it is kinda confusing. I tried your setInstance implementation, nothing changed, it still sends the message but no entity in sight

**Reply by GreatWyrm** [reply]
Huh, that's worked for me.

Do you override the setInstance/teleport functions in your PassiveMob class?
Could also be a viewable issue, do you have any custom viewable rules/alter the default viewable stuff in any way?

**Reply by Tofaa2** [reply]
Nope, none modified, again it worked perfectly fine now it just stopped. I've modified other commands but this wasnt touched at all. I'm gonna try checking if the entity and the player instances are correct but it doesnt work

**Reply by Tofaa2** [reply]
![image](https://user-images.githubusercontent.com/82680183/207965755-c5134c65-2885-4527-ac6c-a8fea2a2ba51.png)
https://hastebin.com/niyuzeqine.java <- Mob class

**Reply by Tofaa2** [reply]
I'm confused, i tried to manually do new ChickenMod().setInstance(instance, position); // ChickenMob is extending PassiveMob with no overrides and nothing changed

**Comment by dev-hydrogen** [comment]
Im curious on the usage of `final PassiveMob mob = mobClass.getConstructor().newInstance();`? (Why not use the `new` keyword instead?)
DId this work before? It seems (with the code provided) that it should not work since you arent passing arguments into the constructor

**Reply by Tofaa2** [reply]
Well the enum simply contains the name and the class needed. It worked before perfectly fine just randomly stopped without its code being altered. It might seem hacky but it is an okay method to dynamically load mobs from an enum (though reflection is used it shouldnt really be that big of a deal, might be slightly slower than new keyword)

**Reply by Tofaa2** [reply]
![image](https://user-images.githubusercontent.com/82680183/207975308-b248cf23-e41a-4bcc-b4e2-0cdb61c75f18.png)

**Reply by Tofaa2** [reply]
By the way in the example above i set the setInstance method inside the constructor, it is normally in this method but i was just testing

**Reply by dev-hydrogen** [reply]
> ![image](https://user-images.githubusercontent.com/82680183/207975308-b248cf23-e41a-4bcc-b4e2-0cdb61c75f18.png)

What is the constructor(s) for ChickenMob?

**Reply by Tofaa2** [reply]
Mentioned above that theyre all superclasses with just the constructor filling info on them. Doing the Minecraft src style approach xD
![image](https://user-images.githubusercontent.com/82680183/207975946-c8e85291-db4f-447f-acbd-d5c1d1db15ab.png)

**Reply by Tofaa2** [reply]
This all used to work fine, i don't think the mob is the issue, perhaps something wrong with viewing?

**Reply by dev-hydrogen** [reply]
If this used to work fine, then what changed? Did you *only* update minestom?

**Reply by Tofaa2** [reply]
I updated other commands but nothing else has changed, only minestom rebuilt by gradle if it did (idk i havent hard changed it). There's really no real reason it shouldnt work iirc idfk that's why im asking haha

**Reply by Tofaa2** [reply]
![image](https://user-images.githubusercontent.com/82680183/207976604-7951bee9-f72c-454b-8044-79c6578faffd.png)
Also this gets loaded and displayed normally

**Comment by Tofaa2** [comment, marked as answer]
Fixed: Fabric client issue for some reason. The server was doing everything correctly
